packages/ui-lib/src/Global/atoms/TopBarBadge.tsx

```diff
@@ -0,0 +1,141 @@
+  a {
+    ${CoreStyle};
+
+    text-decoration: none;
+
+    ${({themeColor}) => themeColor ? css`
+      background-color: transparent;
+      border: 2px solid var(--${themeColor}-9);
+      color: var(--${themeColor}-11);
+    ` : css`
+      background-color: transparent;
+      border: 2px solid var(--info-9);
+      color: var(--info-11);
+    `};
+
+    &:hover {
+      ${DefaultText}{
+        display: none;
+      }
+
+      ${LinkText}{
+        display: inline-block;
+      }
+
+      ${({themeColor}) => themeColor ? css`
+        background-color: var(--${themeColor}-9);
+        border: 2px solid var(--${themeColor}-9);
+        color: var(--white-12);
+      ` : css`
+        background-color: var(--info-9);
+        border: 2px solid var(--info-9);
+        color: var(--white-12);
+      `};
+    }
+  }
+`;
+
+const Container = styled.div<{ready: boolean, minWidth: number}>`
+  ${({ready, minWidth}) => css`
+    
+    visibility: ${ready ? 'visible' : 'hidden'};
+    opacity: ${ready ? '1' : '0'};
+
+    transition: opacity var(--speed-fast) var(--easing-primary-in-out);
+
+    ${LinkText}{
+      display: ${ready ? 'none' : 'inline-block'};;
+    }
+
+    ${DefaultText}, ${LinkText}{
+      /* Required to accurately measure container sizes and ensure hover doesn't resize. */
+      ${ready ? `min-width: ${minWidth}px;` : 'position: absolute;'};
+    }
+  `};
+`;
+
+const TopBarBadge: React.FC<ITopBarBadge> = ({text, color, linkHref, linkTo, linkText}) => {
+  
+  const defaultTextRef = useRef() as React.MutableRefObject<HTMLSpanElement>;
+  const linkTextRef = useRef() as React.MutableRefObject<HTMLSpanElement>;
+
+  const [ready, setReady] = useState<boolean>(false);
+  const [minWidth, setMinWidth] = useState<number>(0);
+
+  useEffect(() => {
+    setReady(false);
+    setTimeout(() => {
+      const defaultWidth = defaultTextRef.current.getBoundingClientRect().width;
+      const linkWidth = linkTextRef.current?.getBoundingClientRect().width || 0;
+      const largestWidth = defaultWidth >= linkWidth ? defaultWidth : linkWidth;
+      console.log(defaultWidth, linkWidth, largestWidth);
+      setMinWidth(Math.ceil(largestWidth));
+      setReady(true);
+    }, 100);
+  }, [defaultTextRef, linkTextRef, text, linkText, setMinWidth, setReady]);
+
+  const defaultTextElement = <DefaultText ref={defaultTextRef}>{text}</DefaultText>;
+  const linkTextElement = linkTo || linkHref ? <LinkText ref={linkTextRef}>{linkText || text}</LinkText> : null;
+  
+  let badgeComponent;
+
+  if(linkTo){
+    badgeComponent = <ContainerLinked themeColor={color}><Link to={linkTo}>{defaultTextElement}{linkTextElement}</Link></ContainerLinked>;
+  } else if(linkHref){
+    badgeComponent = <ContainerLinked themeColor={color}><a href={linkHref}>{defaultTextElement}{linkTextElement}</a></ContainerLinked>;
+  } else {
+    badgeComponent = <ContainerStatic themeColor={color}>{defaultTextElement}</ContainerStatic>;
+  }
+  
```

Comment:
Sure, let's chat through this on Monday. Just to share some thoughts and notes quickly, if only for my own benefit...

I deliberately chose to avoid asking for a width manually. I believe that is likely to cause more problems than it will solves.

I've had a look through the link, which was a good read and opened my mind to a few new things, thank you! I'm not sure I've found anything that will help there as I'm pretty sure it's a font load in issue... so without the time out it usually measure the width just before the custom font drops in meaning the size changes slightly. Definitely one worth chatting through!